modules/monitoring/library/Monitoring/Hook/EventDetailsExtensionHook.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/* Icinga Web 2 | (c) 2017 Icinga Development Team | GPLv2+ */
+
+namespace Icinga\Module\Monitoring\Hook;
+
+/**
+ * Base class for hooks extending the event view of monitored objects
+ *
+ * Extend this class if you want to extend the event view of monitored objects with custom HTML.
+ */
+abstract class EventDetailsExtensionHook
```

Comment:
Please teach this to know about a view, it's module and a final public constructor. Just like the [DetailviewExtensionHook](https://github.com/Icinga/icingaweb2/blob/master/modules/monitoring/library/Monitoring/Hook/DetailviewExtensionHook.php).

Reply:
Eh, forget this about the view. That's not necessary.

Reply:
Why is this resolved? I wrote passing the view is not necessary but the other two are still necessary. The module object may be useful for implementations which require configuration. The constructor should be implemented and final so that we can guarantee that any implementation can be successfully instantiated. (If an implementation needs initialization, it should override method `init()`)